src/Hint/Structure.hs

```diff
@@ -39,6 +39,7 @@ foo = let ~(x:xs) = y in z
 module Hint.Structure(structureHint) where
 
 import Hint.Type
+import Data.Function (on)
```

Comment:
I tend not to like explicit import lists, they involve lots of import editing, especially for things where the module is in base.